src/api/commands/currencyprice/CurrencyPriceAddCommand.ts

```diff
@@ -0,0 +1,66 @@
+import { inject, named } from 'inversify';
+import { validate, request } from '../../../core/api/Validate';
+import { Logger as LoggerType } from '../../../core/Logger';
+import { Types, Core, Targets } from '../../../constants';
+import { CurrencyPriceService } from '../../services/CurrencyPriceService';
+import { RpcRequest } from '../../requests/RpcRequest';
+import { RpcCommandInterface } from '../RpcCommandInterface';
+import { Commands } from '../CommandEnumType';
+import { BaseCommand } from '../BaseCommand';
+import { MessageException } from '../../exceptions/MessageException';
+import { CurrencyPriceParams } from '../../requests/CurrencyPriceParams';
+import { SupportedCurrencies } from '../../enums/SupportedCurrencies';
+
+export class CurrencyPriceAddCommand extends BaseCommand implements RpcCommandInterface<any> {
+
+    public log: LoggerType;
+
+    constructor(
+        @inject(Types.Core) @named(Core.Logger) public Logger: typeof LoggerType,
+        @inject(Types.Service) @named(Targets.Service.CurrencyPriceService) private currencyPriceService: CurrencyPriceService
+    ) {
+        super(Commands.CURRENCYPRICE_ADD);
+        this.log = new Logger(__filename);
+    }
+
+    /**
+     *
+     * @param data
+     * @returns {Promise<Bookshelf.Model<any>>}
+     */
+    @validate()
+    public async execute( @request(RpcRequest) data: RpcRequest): Promise<any> {
+
+        if (data.params[0] !== 'PART' || data.params.length < 2 ) {
+           throw new MessageException('Invalid params');
+        } else {
+
+            const fromCurrency = data.params.shift();
+            const unSupportedCurrencies: any = [];
+            for (const toCurrency of data.params) {
```

Comment:
gather the params in Command, then pass the execution to service..
add service.getCurrencyPrices(from, to[]) or something like that.